libraries/AP_RangeFinder/AP_RangeFinder_BLPing.cpp

```diff
@@ -116,9 +116,9 @@ void PingProtocol::send_message(AP_HAL::UARTDriver *uart, PingProtocol::MessageI
     crc += LOWBYTE(payload_len) + HIGHBYTE(payload_len);
 
     // message id
-    uart->write(LOWBYTE(msg_id));
-    uart->write(HIGHBYTE(msg_id));
-    crc += LOWBYTE(msg_id) + HIGHBYTE(msg_id);
+    uart->write(LOWBYTE(static_cast<uint16_t>(msg_id)));
```

Comment:
implicit cast are wrong
hidden cast are wrong
All cast come with some cost, they shouldn't be hidden. If someone want to use a cast he should do it with good reason, not just because it is convenient. 
With the current macro you could do : 
```
double plop= -2.9;
printf("low: %d, hight %d\n", LOWBYTE(plop), HIGHBYTE(plop));
```
and it will be fine ...
You want to use bit manipulation then you should have unsigned int, if you don't have it, think again and cast if it is safe. In this case, that the right thing to do as msg_id is an enum class of type uint16_t
I am sorry but you are wrong. The will to make things simpler shouldn't sacrifice safety and correctness.